MediaBrowser.Providers/Manager/ItemImageProvider.cs

```diff
@@ -206,7 +206,7 @@ private bool HasImage(BaseItem item, ImageType type)
         /// <param name="backdropLimit">The backdrop limit.</param>
         /// <param name="screenshotLimit">The screenshot limit.</param>
         /// <returns><c>true</c> if the specified item contains images; otherwise, <c>false</c>.</returns>
-        private bool ContainsImages(BaseItem item, List<ImageType> images, TypeOptions savedOptions, int backdropLimit, int screenshotLimit)
+        private bool ContainsImages(BaseItem item, IEnumerable<ImageType> images, TypeOptions savedOptions, int backdropLimit, int screenshotLimit)
```

Comment:
You are enumerating multiple times in this function 😕 `images.Contains()` is called twice.

Reply:
`images.Contains()` was called twice before my changes as well. However I think I can tweak the code to just go over the collection once.

Reply:
Fixed.